satpy/tests/reader_tests/test_fci_l2_nc.py

```diff
@@ -233,152 +256,164 @@ def setUp(self):
 
         expected_global_attributes = {
             'filename': self.seg_test_file,
-            'start_time': datetime.datetime(year=2017, month=9, day=20,
-                                            hour=17, minute=30, second=40),
-            'end_time': datetime.datetime(year=2017, month=9, day=20,
-                                          hour=17, minute=41, second=17),
             'spacecraft_name': 'test_fci_platform',
             'ssp_lon': 0.0,
             'sensor': 'test_fci_data_source',
-            'creation_time': datetime.datetime(year=2017, month=9, day=20,
-                                               hour=12, minute=30, second=30),
             'platform_name': 'test_fci_platform'
         }
         self.assertEqual(global_attributes, expected_global_attributes)
 
-    def test_dataset(self):
+    def test_dataset_without_adef(self):
         """Test the execution of the get_dataset function."""
+        self.fh = FciL2NCSegmentFileHandler(filename=self.seg_test_file, filename_info={}, filetype_info={})
+
         # Checks the correct execution of the get_dataset function with a valid file_key
-        dataset = self.segment_reader.get_dataset(None,
-                                                  {'file_key': 'test_values',
-                                                   'fill_value': -999, 'mask_value': 0, })
-        self.assertTrue(np.allclose(dataset.values, np.ones((100, 10, 8, 6))))
+        dataset = self.fh.get_dataset(make_dataid(name='test_values', resolution=32000),
+                                      {'name': 'test_values',
+                                       'file_key': 'test_values',
+                                       'fill_value': -999, })
+        expected_dataset = self._get_unique_array(range(8), range(6))
+        self.assertTrue(np.allclose(dataset.values, expected_dataset))
         self.assertEqual(dataset.attrs['test_attr'], 'attr')
         self.assertEqual(dataset.attrs['units'], 'test_units')
         self.assertEqual(dataset.attrs['fill_value'], -999)
 
         # Checks the correct execution of the get_dataset function with an invalid file_key
-        invalid_dataset = self.segment_reader.get_dataset(None,
-                                                          {'file_key': 'test_invalid',
-                                                           'fill_value': -999, 'mask_value': 0})
+        invalid_dataset = self.fh.get_dataset(make_dataid(name='test_invalid', resolution=32000),
+                                              {'name': 'test_invalid',
+                                               'file_key': 'test_invalid',
+                                               'fill_value': -999, })
         # Checks that the function returns None
         self.assertEqual(invalid_dataset, None)
 
+        # Checks that no AreaDefintion is implemented
+        self.assertRaises(NotImplementedError, self.fh.get_area_def, None)
```

Comment:
you could use pytest's conterpart here if you wanted too.
```python
with pytest.raises(NotImplementedError):
    self.fh.get_area_def(None)
```

Reply:
Fixed.